src/Eventso.Subscription/Observing/DeadLetter/DeadLetterQueue.cs

```diff
@@ -0,0 +1,58 @@
+using System;
+using System.Collections.Generic;
+
+namespace Eventso.Subscription.Observing.DeadLetter
+{
+    public sealed class DeadLetterQueue : IDeadLetterQueue
+    {
+        private static readonly Dictionary<object, IReadOnlySet<string>> EmptyDeadLetters = new(); 
+        
+        private readonly object _lockObject = new();
+
+        private Dictionary<object, IReadOnlySet<string>> _knownDeadLetters;
```

Comment:
I think Set is wrong here. We shouldn't think for the client. When he add same strings - its ok: all his strings will be added. List is enought and simple

Reply:
true, done